torch/csrc/autograd/functions/utils.h

```diff
@@ -26,6 +26,12 @@ inline tensor_list as_tensor_list(Args&& ... args) {
 }
 
 
+/**
+ * Create a Variable which records the result of applying the function ctr to
+ * inputs (having produced outputs), if necessary.  If any of the inputs are
+ * volatile, this function discards the input graph and returns just the
+ * outputs.
+ */
```

Comment:
It wraps the tensor outputs in Variables, and if necessary uses the function constructor to create grad_fn for them (it doesn't discard anything).